src/node_api.cc

```diff
@@ -1090,14 +1094,23 @@ napi_status napi_define_properties(napi_env env,
   for (size_t i = 0; i < property_count; i++) {
     const napi_property_descriptor* p = &properties[i];
 
-    v8::Local<v8::Name> name;
-    CHECK_NEW_FROM_UTF8(env, name, p->utf8name);
+    v8::Local<v8::Name> property_name;
+
+    if (p->utf8name != nullptr) {
```

Comment:
Spent a bit doing this now. I want this helper to live in the `v8impl` namespace which is located above the helper macros like `CHECK_NEW_FROM_UTF8` in the source. Looks like we'll need to move the helper macros above `v8impl`. Any reservations?